lib/adagrams.rb

```diff
@@ -0,0 +1,70 @@
+# Wave 1 [Question 1 - Creating Data Structure]
+
+def draw_letters
+    pool_of_letters = [
+    "A", "A", "A", "A", "A", "A", "A", "A", "A", "B", "B", "C", "C", "D", "D", "D", "D", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "F", "F", "G", "G", "G", "H", "H", "I", "I", "I", "I", "I", "I", "I", "I", "I", "J", "K", "L", "L", "L", "L", "M", "M", "N", "N", "N", "N", "N", "N", "O", "O", "O", "O", "O", "O", "O", "O", "P", "P", "Q", "R", "R", "R", "R", "R", "R", "S", "S", "S", "S", "T", "T", "T", "T", "T", "T", "U", "U", "U", "U", "V", "V", "W", "W", "X", "Y", "Y", "Z"
+]
+    my_letters = []
+    random_letter = pool_of_letters.shuffle
+    10.times do
+        my_letters.push(random_letter[-1])
```

Comment:
Just to make it clear what's going on: You have the `pool_of_letters` array, and then shuffle it, and put the shuffled array into a variable called `random_letter`. Then, you do this ten times: you push into your array `my_letters` the value of the item that is last in the `random_letter` array (the thing at index of `-1` is the last item). And then you call `random_letter.pop`, which removes the last item from the array. This strategy is great! It just feels a little unreadable. I think that the name `random_letter` feels inaccurate, since it doesn't represent one single random letter, but the array of the shuffled pool of letters. Also, instead of maybe using the syntax `random_letter[-1]`, even though it's very short and concise, it might be more readable to use `random_letter.last`.

There are also interesting things that `.pop` does that you'll learn about in the future :)